db.json

```diff
@@ -0,0 +1,84 @@
+{
+  "projects": [
+    {
+      "id": "p-001",
+      "label": "project 1",
+      "description": "This is the first project"
+    },
+    {
+      "id": "p-002",
+      "label": "project 2",
+      "description": "This is the second project"
+    }
+  ],
+
+  "projects-data": [
+    {
+      "id": "p-001",
+      "label": "Project 1",
+      "description": "This is the first project",
+      "pages": [
+        "page-001-001",
+        "page-001-002",
+        "page-001-003"
+      ]
+    },
+    {
+      "id": "p-002",
+      "label": "Project 2",
+      "description": "This is the second project",
+      "pages": [
+        "page-002-001",
+        "page-002-002"
+      ]
+    }
+  ],
+
+  "pages-data": [
+    {
+      "id": "page-001-001",
+      "label": "First page",
+      "description": "First page of first project",
+      "widgets": []
+    },
+    {
+      "id": "page-001-002",
+      "label": "Second page",
+      "description": "Second page of first project",
+      "widgets": []
+    },
```

Comment:
should there be data for "page-001-003" or is this intentional to show what happens when navigating to a page that has no data?

Reply:
1. All the api-services methods are returning observables. https://github.com/dasch-swiss/dsp-cpe/blob/routing/src/app/repository/cpe-api.service.ts We should talk about the use of https://github.com/Effect-TS/effect before implementing stuff deeper here. Also there is a lot to consider which is definitely not covered by this PR.
The mock projects component and the pages component still consume promises provided by the repositories. We should look at the structure/architecture before changing that. They will probably be removed anyways or changed dramatically. So changing them to observables in this PR makes no sense to me.
2. Yes this is intentional. I like to catch the case of inconsistent data. It defaults to the first page of the project.